### PR TITLE
[WIP] Port database

### PR DESCRIPTION
**Pull request by Dragneel7**
- Create Models file
- Create Redis Reminder client

### CODE REVIEW

**Comment by Dragneel7** [comment]
- Changes remaining for index.html : 
1.) Convert WKB to GeoJSON format for rendering the map.
2.) Allow limit option and view inactive option on getCampaginFunction

**Comment by Dragneel7** [comment]
Modifications to Create/Edit Campaign:
1.) find a way to create to extract team and geometry data from the form data.
2.) create a popup to allow campaign_creator to choose from pre-existing team or create a new team.
**Overall Testing of the Application**

File: deployment/docker/requirements.txt [hunk 1]
```diff
@@ -15,4 +15,6 @@ flask-restful==0.3.6
 PyYAML
 numpy
 Flask-Script
-simplekml==1.3.0
\ No newline at end of file
+simplekml==1.3.0
+psycopg2
```
Comment (timlinux) [inline deployment/docker/requirements.txt]: use specific versions please

File: flask_project/campaign_manager/models/models.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
+import os
+
+from sqlalchemy import create_engine, Column, Integer, String,\
```
Comment (timlinux) [inline flask_project/campaign_manager/models/models.py]: from sqlalchemy import (
    create_engine, 
    Column, 
    Integer, 
    String,
    etc.
)

File: flask_project/campaign_manager/models/models.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            Table, Text, ForeignKey, Date, DateTime, BOOLEAN
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+from geoalchemy2 import Geometry
+
+# change the db-name, db-owner and db-password
+engine = create_engine('postgresql://surya:hello@localhost/fieldcampaigner',echo=True)
```
Comment (timlinux) [inline flask_project/campaign_manager/models/models.py]: Use env vars rather for this kind of stuff

File: flask_project/campaign_manager/models/models.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+Base = declarative_base()
+
+
+"""
+  Association Tables to create many-to-many relation between models.
+"""
+adminAssociations = Table('adminAssociations', Base.metadata,
```
Comment (timlinux) [inline flask_project/campaign_manager/models/models.py]: admin_associations

File: flask_project/campaign_manager/models/models.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+"""
+    Users model: stores the user data
+"""
+class User(Base):
+    __tablename__ = 'user'
+
+    id = Column('id_user', Integer, primary_key=True, autoincrement=True)
+    osm_user_id = Column(String(20), unique=True, nullable=False)
+    email = Column(String(40))
+    campaign_creator = relationship('Campaign', back_populates='creator',
```
Comment (timlinux) [inline flask_project/campaign_manager/models/models.py]: pull left

File: flask_project/campaign_manager/models/models.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+                lazy=True)
+    geometry = Column(Geometry('POLYGON'))
+    taskBoundaries = relationship('TaskBoundary', back_populates='campaign',
+                lazy=True)
+    name = Column(String(20), unique=True, nullable=False)
+    description = Column(String(200))
+    start_date = Column(Date(), nullable=False)
+    end_date = Column(Date(), nullable=False)
+    create_on = Column(DateTime(), nullable=False)
+    link_to_OpenMapKit = Column(BOOLEAN(), default=False)
```
Comment (timlinux) [inline flask_project/campaign_manager/models/models.py]: pythonic naming = link_to_openmapkit

**Comment by timlinux** [comment]
Can you make sure all tests pass please?

**Comment by Dragneel7** [comment]
@timlinux I have successfully integrated the models(CRUD) tests with travis. I'll squash the commit and refactor the code according to kartoza guidelines.

File: deployment/docker/requirements.txt (hunk 1 above)
Comment (Dragneel7) [reply deployment/docker/requirements.txt]: added specific versions for requirements.

**Comment by Dragneel7** [comment]
#433 New pull request.

**Pull request closed without merging**